semantic/utils.go

```diff
@@ -0,0 +1,55 @@
+package semantic
+
+import "github.com/influxdata/flux/ast"
+
+func MergeArrowFunction(a, b *FunctionExpression) *FunctionExpression {
+	fn := a.Copy().(*FunctionExpression)
+
+	aExp, aOK := a.Body.(Expression)
+	bExp, bOK := b.Body.(Expression)
+
+	if aOK && bOK {
+		fn.Body = &LogicalExpression{
+			Operator: ast.AndOperator,
+			Left:     aExp,
+			Right:    bExp,
+		}
+		return fn
+	}
+
+	// TODO(nathanielc): This code is unreachable while the current PushDownRule Match function is inplace.
```

Comment:
This code is now reachable. However, storage can only support expressions as predicates. I.e no return statements and such. Since this is used in an actual rewrite rule [https://github.com/influxdata/flux/pull/159/files#diff-fa60b7ee55bcb82c7acd49cc6bca844aR204](url), I think there should be a check (in the rule) that the filter is an expression, and if not, don't perform the rewrite.

Reply:
I will at the very least make sure that we only push down filters that are simple expressions and not statements.